packages/augur-core/src/contracts/trading/ERC20Proxy.sol

```diff
@@ -0,0 +1,126 @@
+pragma solidity 0.5.15;
+
+import 'ROOT/libraries/CloneFactory.sol';
+import 'ROOT/libraries/token/IERC20.sol';
+import 'ROOT/libraries/token/IERC1155.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+
+
+contract ERC20ProxyNexus {
+    using SafeMathint256 for uint256;
+
+    address public target1155;
+    address proxyToClone;
+
+    // tokenId -> owner -> spender -> allowance
+    // The spender is whoever called the proxy contract, which is msg.sender in the proxy and _caller here in the nexus.
+    mapping (uint256 => mapping (address => mapping (address => uint256))) public allowances;
+    // tokenId -> proxy contract
+    mapping (uint256 => address) public proxies;
+
+    constructor(address _proxyToClone, address _target1155) public {
+        proxyToClone = _proxyToClone;
+        target1155 = _target1155;
+        // This contract will be able to do anything with the user's 1155 tokens.
+        IERC1155(target1155).setApprovalForAll(address(this), true);
```

Comment:
The way this actually has to work is each user will call `setApprovalForAll` for this contract.

Reply:
OH duh!

Reply:
Looking at the OZ ERC1155 implementation, it seems that the user will have to call the ShareToken contract's setApprovalForAll themselves. So there's nothing to be done here. It'll just fail if the user tries to do anything requiring approval.